src/darkpool/Darkpool.sol

```diff
@@ -815,6 +816,9 @@ contract Darkpool is Initializable, Ownable2Step, Pausable {
 
         // 4. Commit the note into the merkle tree
         merkleTree.insertLeaf(statement.noteCommitment, hasher);
+
+        // 5. Emit the event
+        emit NotePosted(statement.noteCommitment);
```

Comment:
Would have liked to add a `commitNote()` helper that inserts + emits` but that pushes us over the contract size limit.